src/ssl.c

```diff
@@ -29194,26 +29211,67 @@ int wolfSSL_X509_PUBKEY_set(WOLFSSL_X509_PUBKEY **x, WOLFSSL_EVP_PKEY *key)
     switch (key->type) {
 #ifndef NO_RSA
     case EVP_PKEY_RSA:
-        pk->algor->algorithm= wolfSSL_OBJ_nid2obj(NID_rsaEncryption);
+        pval = NULL;
+        ptype = V_ASN1_NULL;
+        pk->pubKeyOID = RSAk;
         break;
+    #ifdef WC_RSA_PSS
+    /* TODO: Encode PSS parameters */
+    //case EVP_PKEY_RSA_PSS:
```

Comment:
I don't see `EVP_PKEY_RSA_PSS` defined... for now can we remove this TODO? Otherwise add `EVP_PKEY_RSA_PSS` and the case, but return error if used. Also prefer C style multiline comments `/* */`.